### ISSUE TITLE
Allow continued use of some shortcuts within a textbox: `imap`?

### ISSUE THREAD

**Issue by GKFX** [philc/vimium#2904]
If I focus a textbox, it obviously makes sense for Vimium not to interpret <kbd>A</kbd> or whatever as a command. However, if I've mapped <kbd>ctrl-g</kbd> to a command, it seems reasonable for Vimium to continue to capture that keystroke, even within a textbox, rather than having Chrome's search bar come up.

Introducing an `imap` command seems like the most comprehensive solution, since that would allow a small number of mappings to be configured to work in text while the rest remain disabled. A "`nimap`"(normal-and-insert-map) would also be useful as it would avoid having to specify these commands twice.

This is prompted largely by the existence of auto-focusing textareas. Facebook Messenger in particular refocuses the textarea even after clicking off it into the chat.

**Comment by lydell** [comment]
For inspiration, VimFx allowed shortcuts to be triggered in text inputs by prefixing them with [`<late>`](https://github.com/akhodakivskiy/VimFx/blob/99402ebdba8744083945c900cc024d91b62e2fe7/documentation/shortcuts.md#force).

**Comment by GKFX** [comment]
The title of this issue has just been changed to "Handle non-printable Vimium key mappings in insert mode". I'm not sure that's wise; what about `<c-a>`? That's something that most users would probably prefer to not be remapped in insert mode. Introducing a distinct `imap` command allows users to make that choice for themselves.

**Comment by smblott-github** [comment]
The `imap` (or `<force>`) ideas would work, but they require the user to do a lot of configuration for a fairly infrequent scenario.  It may be too niche to be worth the extra complexity.

Also, I simply wouldn't be able to remember all of the special-case bindings.

(And I get @GKFX's point about unilaterally having Vimium handle all non-printable keystrokes.)

An alternative might be to implement this:

    map <c-]> passNextKey normal

This fires up a new normal-mode instance, with the regular key bindings.  So, to visit the tab on the right... `<c-]>K`.

This syntax exists, it just doesn't work correctly in insert mode.  A first cut at a fix is [here](4e6b9c1a623e3270ed9aae5c0c2bf5fc5a2e195a).

This works too...

    map <c-]> passNextKey normal count=999999

**Comment by GKFX** [comment]
I see your point about extra configuration being a barrier to this being used, but pressing `<c-]>` is exactly as hard as just pressing esc to unfocus the text box. If I know that a command is set to always work, then I can just press it with confidence, rather than thinking “am I in a text box; what do I do?”, which wastes time forever, while configuration only takes up my time once.  Either way, thank you for considering the request.

**Comment by smblott-github** [comment]
> is exactly as hard as just pressing esc to unfocus the text box.

Yes, except that (in a correct implementation) when you're done you should end up with the focus back in the original input.  (4e6b9c1a623e3270ed9aae5c0c2bf5fc5a2e195a isn't enough in this regard.)
